crates/data-generator/src/blocks/json.rs

```diff
@@ -0,0 +1,132 @@
+use std::collections::BTreeMap;
+
+use serde::Deserialize;
+use thiserror::Error;
+
+use super::model::{Block, BlockProperty, BlockState, BlockStateProperty, Id};
+
+#[derive(Debug, Deserialize)]
+struct SerdeId(u32);
+
+#[derive(Debug, Deserialize)]
+struct SerdeBlock {
+    #[serde(default)]
+    properties: BTreeMap<String, Vec<String>>,
+    states: Vec<SerdeState>,
+}
+
+#[derive(Debug, Deserialize)]
+struct SerdeState {
+    id: SerdeId,
+    #[serde(default)]
+    default: bool,
+    #[serde(default)]
+    properties: BTreeMap<String, String>,
+}
+
+#[derive(Debug, Error)]
+pub enum ParseError {
+    #[error("malformed JSON data: {0}")]
+    Json(#[from] serde_json::Error),
+}
+
+pub(super) fn parse_block_registry(json_data: String) -> Result<Vec<Block>, ParseError> {
+    let registry: BTreeMap<String, SerdeBlock> = serde_json::from_str(&json_data)?;
+
+    fn block_property(
+        block_name_pc: String,
+        prop_name_sc: String,
+        prop_variants: Vec<String>,
+    ) -> BlockProperty {
+        let prop_variants = prop_variants
+            .iter()
+            .map(|prop_variant| transform_to_pascal_case(prop_variant))
```

Comment:
```suggestion
            .map(transform_to_pascal_case)
```

The same in all similar places

Reply:
Should be done; look in worktree at [d8fa342](https://github.com/rjacraft/rjacraft/tree/feature/data-generator). Quickly skimming through the code base, I didn't find anything.